### PR TITLE
Dev loll

### PR DESCRIPTION
**Pull request by lollcat**
- improve stability of error calculation in many well
- update run instructions for gmm and many well
- various other small changes

### CODE REVIEW

File: experiments/many_well/README.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ The following commands can each be used to train the methods from the paper:
 python experiments/many_well/run.py -m training.seed=0,1,2 training.use_buffer=True training.prioritised_buffer=True 
 
 # FAB without the prioritised buffer.
-python experiments/many_well/run.py -m training.seed=0,1,2 fab.loss_type=fab_p2_over_q_alpha_2_div 
+python experiments/many_well/run.py -m training.seed=0,1,2 fab.loss_type=fab_alpha_div
```
Comment (lollcat) [inline experiments/many_well/README.md:15]: This name changed, need to do the matching change inside the README

File: experiments/many_well/ground_truth_higher_order_marginals.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import numpy as np
```
Comment (lollcat) [inline experiments/many_well/ground_truth_higher_order_marginals.py:1]: Make a histogram showing the frequency of how often higher order marginals are sampled from. Not used in the paper, but nice to have in the repo if we come back to this problem.

File: experiments/setup_run.py [hunk 3]
```diff
@@ -188,7 +188,7 @@ def setup_model(cfg: DictConfig, target: TargetDistribution) -> FABModel:
             alpha=cfg.fab.alpha,
             p_target=p_target,
             n_outer=1,
-            epsilon=1.0,
```
Comment (lollcat) [inline experiments/setup_run.py:191]: Previously this was left as 1. But it should be configurable.

File: fab/target_distributions/many_well.py [hunk 4]
```diff
@@ -108,8 +108,9 @@ def performance_metrics(self, samples: torch.Tensor, log_w: torch.Tensor,
             log_w = log_w[:n_vals_per_split*n_runs]
             log_w = torch.stack(log_w.split(n_runs), dim=-1)
             # Check accuracy in estimating normalisation constant.
-            Z_estimate = torch.exp(torch.logsumexp(log_w, axis=-1) - np.log(log_w.shape[-1]))
-            MSE_Z_estimate = torch.mean(torch.abs((Z_estimate - self.Z) / self.Z))
+            log_Z_estimate = torch.logsumexp(log_w, axis=-1) - np.log(log_w.shape[-1])
```
Comment (lollcat) [inline fab/target_distributions/many_well.py:111]: The previous way I calculated this was only stable for float 64 (broke for float32)

File: fab/core.py [hunk 5]
```diff
@@ -215,7 +215,7 @@ def save(self,
              path: "str"
              ):
         """Save FAB model to file."""
-        torch.save({'flow': self.flow._nf_model.state_dict(),
```
Comment (lollcat) [inline fab/core.py:218]: flow is already a torch module, so we can save and load it directly

**Review by VincentStimper — APPROVED** [review]
Looks great, thanks!

**Pull request merged**